var/spack/repos/builtin/packages/miopen-hip/package.py

```diff
@@ -36,11 +42,21 @@ class MiopenHip(CMakePackage):
         depends_on('rocblas@' + ver, type='link', when='@' + ver)
         depends_on('rocm-device-libs@' + ver, type='link', when='@' + ver)
 
+    def setup_build_environment(self, env):
+        if '@3.9.0:' in self.spec:
+            lib_dir = self.spec['zlib'].prefix.lib
```

Comment:
```suggestion
            lib_dir = self.spec['zlib'].libs.directories[0]
```
The library may be in prefix.lib or prefix.lib64 depending on the system